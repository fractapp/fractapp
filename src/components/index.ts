export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
