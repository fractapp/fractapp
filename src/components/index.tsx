export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
