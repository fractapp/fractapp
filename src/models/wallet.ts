import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
