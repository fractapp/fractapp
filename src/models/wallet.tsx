import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
