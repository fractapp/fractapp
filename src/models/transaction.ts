import {Currency} from './wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export class Transaction {
  id: string;
  member: string;
  currency: Currency;
  value: number;
  txType: TxType;
  timestamp: number;
  fee: number;
  usdValue: number;
  usdFee: number;
  status: TxStatus;

  constructor(
    id: string,
    member: string,
    currency: Currency,
    txType: TxType,
    timestamp: number,
    value: number,
    usdValue: number,
    fee: number,
    usdFee: number,
    status: TxStatus,
  ) {
    this.id = id;
    this.member = member;
    this.currency = currency;
    this.value = value;
    this.txType = txType;
    this.timestamp = timestamp;
    this.fee = fee;
    this.usdValue = usdValue;
    this.usdFee = usdFee;
    this.status = status;
  }
}
