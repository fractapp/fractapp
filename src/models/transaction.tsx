import { ImageSourcePropType } from "react-native";
import { Currency } from "./wallet";

export enum TxType {
    None = 0,
    Sent,
    Received
}

export class Transaction {
    id: string;
    member: string;
    currency: Currency;
    value: number;
    txType: TxType;
    date: Date;

    usdValue: number;
    symbol: string;
    logo: ImageSourcePropType;

    constructor(id: string, member: string, currency: Currency, txType: TxType, date: Date, value: number) {
        this.id = id;
        this.member = member;
        this.currency = currency;
        this.value = value;
        this.txType = txType;
        this.date = date;
        //TODO get state
        let rate = 10
        this.usdValue = value * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
