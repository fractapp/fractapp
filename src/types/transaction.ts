import {Currency} from './/wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

/**
 * Transaction action type
 * @category Models
 */
export enum TxAction {
  Transfer,
  StakingReward,
  StakingCreateWithdrawalRequest,
  StakingWithdrawn,
  StakingOpenDeposit,
  StakingAddAmount,
  ConfirmWithdrawal,
  UpdateNomination
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export type Transaction = {
  id: string;
  hash: string;
  userId: string | null;

  address: string;
  currency: Currency;
  action: TxAction,
  txType: TxType;
  timestamp: number;

  value: number;
  planckValue: string;
  usdValue: number;
  fullValue: string;

  fee: number;
  planckFee: string;
  usdFee: number;

  status: TxStatus;
};


/**
 * @category Models
 */
export type ApiTransaction = {
  id: string;
  hash: string;
  currency: Currency;

  from: string;
  userFrom: string;

  action: TxAction,

  to: string;
  userTo: string;

  value: string;
  fee: string;
  price: number

  timestamp: number;
  status: TxStatus;
};

