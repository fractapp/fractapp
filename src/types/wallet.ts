import { ImageSourcePropType } from 'react-native';
import { AccountType, Network } from 'types/account';
import { Transaction, TxAction } from 'types/transaction';

/**
 * @category Models
 */
export enum Currency {
  DOT = 0,
  KSM,
}


/**
 * get network by currency
 * @category Models
 */

export function getNetwork(currency: Currency): Network {
  switch (currency) {
    case Currency.DOT:
      return Network.Polkadot;
    case Currency.KSM:
      return Network.Kusama;
    default:
      return Network.Polkadot;
  }
}


/**
 * get logo for wallet
 * @category Models
 */

export function getFullCurrencyName(currency: Currency) {
  let name = '';
  switch (currency) {
    case Currency.DOT:
      name = 'Polkadot';
      break;
    case Currency.KSM:
      name = 'Kusama';
      break;
    default:
      throw new Error('invalid currency');
  }
  return name;
}

export function getColor(currency: Currency) {
  let color = '';
  switch (currency) {
    case Currency.DOT:
      color = '#E6007A';
      break;
    case Currency.KSM:
      color = '#434C74';
      break;
    default:
      throw new Error('invalid currency');
  }
  return color;
}

export function getLogo(currency: Currency) {
  let logo: ImageSourcePropType;
  switch (currency) {
    case Currency.DOT:
      logo = require('assets/img/dot.png');
      break;
    case Currency.KSM:
      logo = require('assets/img/kusama.png');
      break;
    default:
      throw new Error('invalid currency');
  }
  return logo;
}

/**
 * check logo without border
 * @category Models
 */
export function withoutBorder(currency: Currency): boolean {
  return currency === Currency.DOT || currency === Currency.KSM;
}

/**
 * get symbol for wallet
 * @category Models
 */
export function getSymbol(currency: Currency) {
  let symbol = '';
  switch (currency) {
    case Currency.DOT:
      symbol = 'DOT';
      break;
    case Currency.KSM:
      symbol = 'KSM';
      break;
    default:
      throw new Error('invalid currency');
  }
  return symbol;
}

/**
 * string to currency
 * @category Models
 */
export function toCurrency(currency: string): Currency {
  switch (currency) {
    case 'DOT':
      return Currency.DOT;
    case 'KSM':
      return Currency.KSM;
    default:
      throw new Error('invalid currency');
  }
}


/**
 * currency to string
 * @category Models
 */
export function fromCurrency(currency: Currency): string {
  switch (currency) {
    case Currency.DOT:
      return 'DOT';
    case Currency.KSM:
      return 'KSM';
    default:
      throw new Error('invalid currency');
  }
}

/**
 * filter Tx by wallet
 * @category Models
 */
export function filterTxsByAccountType(txs: Array<Transaction>, action: AccountType): Array<Transaction>  {
  switch (action) {
    case AccountType.Main:
      return txs.filter((tx) => tx.action === TxAction.Transfer || tx.action === TxAction.Staking || tx.action === TxAction.StakingWithdrawn);
    case AccountType.Staking:
      return txs.filter((tx) => tx.action === TxAction.Staking || tx.action === TxAction.StakingWithdrawn || tx.action === TxAction.StakingReward);
    default:
      throw new Error('invalid action type');
  }
}
