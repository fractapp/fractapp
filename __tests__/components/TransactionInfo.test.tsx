import React from 'react';
import renderer from 'react-test-renderer';
import { TransactionInfo } from 'components';
import { Currency } from 'models/wallet'
import { Transaction, TxType } from 'models/transaction';

it('Test currenyc=polkadot && txType=None && member length < 13', () => {
    const tree = renderer
        .create(<TransactionInfo transaction={new Transaction("1", "address#1", Currency.Polkadot, TxType.None, new Date("12-12-2020"), 10, 10)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Test currenyc=kusams && txType=None && member length < 13', () => {
    const tree = renderer
        .create(<TransactionInfo transaction={new Transaction("2", "address#2", Currency.Kusama, TxType.None, new Date("01-12-2020"), 5, 3)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Test member length > 13', () => {
    const tree = renderer
        .create(<TransactionInfo transaction={new Transaction("1", "1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE", Currency.Polkadot, TxType.None, new Date("12-12-2020"), 10, 10)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Test txType=Sent', () => {
    const tree = renderer
        .create(<TransactionInfo transaction={new Transaction("1", "1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE", Currency.Polkadot, TxType.Sent, new Date(), 10, 10)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Test txType=Received', () => {
    const tree = renderer
        .create(<TransactionInfo transaction={new Transaction("1", "1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE", Currency.Polkadot, TxType.Received, new Date(), 10, 10)} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


