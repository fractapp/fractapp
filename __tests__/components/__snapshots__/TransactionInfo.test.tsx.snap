// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test currenyc=kusams && txType=None && member length < 13 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/kusama.png",
          }
        }
        style={
          Object {
            "borderRadius": 75,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "marginLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 17,
                    "fontStyle": "normal",
                  },
                  Object {
                    "color": "#888888",
                    "textAlign": "left",
                  },
                ]
              }
            >
              
              5
               
              KSM
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          address#2
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test currenyc=polkadot && txType=None && member length < 13 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/dot.png",
          }
        }
        style={
          Object {
            "borderRadius": 75,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "marginLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 17,
                    "fontStyle": "normal",
                  },
                  Object {
                    "color": "#888888",
                    "textAlign": "left",
                  },
                ]
              }
            >
              
              10
               
              DOT
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          address#1
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test member length > 13 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/dot.png",
          }
        }
        style={
          Object {
            "borderRadius": 75,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "marginLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 17,
                    "fontStyle": "normal",
                  },
                  Object {
                    "color": "#888888",
                    "textAlign": "left",
                  },
                ]
              }
            >
              
              10
               
              DOT
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          1exaAg2VJRQby...3zo23R2T6EGdE
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test txType=Received 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/dot.png",
          }
        }
        style={
          Object {
            "borderRadius": 75,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "marginLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 17,
                    "fontStyle": "normal",
                  },
                  Object {
                    "color": "#84D371",
                    "textAlign": "left",
                  },
                ]
              }
            >
              +
              10
               
              DOT
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          1exaAg2VJRQby...3zo23R2T6EGdE
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test txType=Sent 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/dot.png",
          }
        }
        style={
          Object {
            "borderRadius": 75,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "marginLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 17,
                    "fontStyle": "normal",
                  },
                  Object {
                    "color": "#F45252",
                    "textAlign": "left",
                  },
                ]
              }
            >
              -
              10
               
              DOT
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 17,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          1exaAg2VJRQby...3zo23R2T6EGdE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
